src/main/java/picard/sam/DownsampleSam.java

```diff
@@ -212,7 +220,45 @@ protected int doWork() {
         }
 
         final SamReader in = SamReaderFactory.makeDefault().referenceSequence(REFERENCE_SEQUENCE).open(SamInputResource.of(INPUT));
-        final SAMFileWriter out = new SAMFileWriterFactory().makeSAMOrBAMWriter(in.getFileHeader(), true, OUTPUT);
+        final SAMFileHeader header = in.getFileHeader().clone();
+
+        if (STRATEGY == Strategy.ConstantMemory || STRATEGY == Strategy.Chained) {
+            //if running using ConstantMemory or Chained strategy, need to check if we have previously run with the same random seed
+            //collect previously used seeds
+            final Integer userSeed = RANDOM_SEED;
+            final Set<Integer> previousSeeds = new HashSet<>();
+            for (final SAMProgramRecord pg : header.getProgramRecords()) {
+                if (pg.getProgramName() != null && pg.getProgramName().equals(PG_PROGRAM_NAME)) {
+                    final String previousSeedString = pg.getAttribute(RANDOM_SEED_TAG);
+                    if (previousSeedString == null) {
+                        /* The previous seed was not recorded.  In this case, the current seed may be the same as the previous seed,
+                        so we will change it to a randomly selected seed, which is very likely to be unique
+                         */
+                        RANDOM_SEED = new Random(pg.hashCode()).nextInt();
```

Comment:
This hash is effectively random since it's the Object.hashCode(), right? but it gives the illusion of being deterministic. 

I think it should be as deterministic as possible, but am not sure what here can be used as a seed. perhaps the collection of PU fields from the RG tags? not sure how you'd like to resolve this, 
but I don't like hidden randomness.

Reply:
`SAMProgramRecord` has a hashCode method that avoids dropping into `Object.hashCode()`, so I believe this should be deterministic.